google/cloud/testing_util/command_line_parsing.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_COMMAND_LINE_PARSING_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_COMMAND_LINE_PARSING_H
+
+#include "google/cloud/version.h"
+#include "absl/types/optional.h"
+#include <chrono>
+#include <cinttypes>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace testing_util {
+
+std::int64_t constexpr kKiB = 1024;
+std::int64_t constexpr kMiB = 1024 * kKiB;
+std::int64_t constexpr kGiB = 1024 * kMiB;
+std::int64_t constexpr kTiB = 1024 * kGiB;
+
+std::int64_t constexpr kKB = 1000;
+std::int64_t constexpr kMB = 1000 * kKB;
+std::int64_t constexpr kGB = 1000 * kMB;
+std::int64_t constexpr kTB = 1000 * kGB;
+
+/// Parse a string as a byte size, with support for unit suffixes.
+std::int64_t ParseSize(std::string const& val);
+
+/**
+ * Parse a string as a byte size, with support for unit suffixes.
+ *
+ * The size must be small enough for an in-memory buffer.
+ */
+std::size_t ParseBufferSize(std::string const& val);
+
+/// Parse a string as a duration with support for hours (h), minutes (m), or
+/// second (s) suffixes.
+std::chrono::seconds ParseDuration(std::string const& val);
+
+/// Parse a string as a boolean, with a default value if the string is empty.
```

Comment:
I'm not sure "a default value" is the best term for "the not-present value".  Either way, it is also the result if the boolean fails to parse, whether or not the string is empty.